vignettes/cookbook.Rmd

```diff
@@ -11,92 +11,630 @@ vignette: >
 
-```{r datasets, echo = FALSE}
-cat("",
-    dir(path = "./../data/", pattern = "\\.RData$") %>%
-      str_replace(".RData$", "") %>%
-      paste(collapse = "\n "))
+* `forcats::fct_reorder()` and `forcats::fct_reorder2()` are useful to automatically order of factors within `mapping`, which establishes legend order. (See also [the relevant R4DS chapter](https://r4ds.had.co.nz/factors.html#modifying-factor-order).)
+
+* functions from the `scales` package, such as `scales::percent()` and `scales::label_percent()`, which reformat numbers into attractive formats.
+
+* ggplot2 scale functions, such as `ggplot2::scale_x_continuous()`, which allow for control over the extent and breaks of each axis. 
+
+#### Example code
+
+```{r line, fig.asp=4.5/(670/72), message=FALSE}
+# Get example dataset, and create a relative GRP field
+df <- peer_grp %>% 
+  group_by(area) %>% 
+  mutate(indexed = grp / first(grp) - 1)
+
+
+# Create line plot
+p <- ggplot(
+  data = df,
+  mapping = aes(
+    x = year,
+    y = indexed,
+    color = fct_reorder2(area, year, indexed),  # Order legend by last `indexed` value in last `year`
+    label = scales::percent(indexed, accuracy = 1)  # Format labels as percents with no decimals
+  )
+) +
+  
+  # Add geoms, making sure to keep recessions under lines
+  geom_recessions(ymin = 0, ymax = .5) +
+  geom_line(lineend = "round") +
+  geom_text_lastonly() +
+  
+  # Establish an attractive Y scale
+  scale_y_continuous(
+    labels = scales::label_percent(accuracy = 1),
+    expand = expansion(add = 0)  # No padding beyond specified limits
+  ) +
+  
+  # Establish an attractive X scale
+  scale_x_continuous(
+    breaks = seq(from = min(df$year), to = max(df$year), by = 2),  # Set breaks to every other year
+    expand = expansion(add = c(0.5, 1.25))  # Set custom padding for L/R sides of axis 
+  ) +
+  
+  # Apply CMAP theme and color palette
+  theme_cmap(
+    overrides = list(margin_panel_r = 3)  # Shrink distance btwn gridlines ending and plotting extent
+  ) +
+  cmap_color_discrete(palette = "legislation")
+  
+
+# Create final plot
+finalize_plot(
+  p,
+  title = "Cumulative growth in real gross regional product, 2001-19",
+  caption = "Note: Index year 2001.<br><br>
+    Source: Chicago Metropolitan Agency for Planning analysis of U.S. Bureau
+    of Economic Analysis and Economic Modeling Specialists International data
+    (Emsi 2020.4).",
+  
+  # Tweak plot layout
+  height = 4.5,
+  title_width = 2.6,
+  overrides = list(
+    margin_plot_r = 0,  # Eliminate distance btwn plot and right side of image
+    margin_legend_b = 25  # Larger than usual gap between legend and plot
+  )
+)
 ```
 
+#### Original graphic
 
-## Sample plots
+<img src="https://www.dropbox.com/s/xol70o61citlpjj/Cumulative_Growth-01.png?raw=1" width="100%" style="display: block; margin: auto;"/>
 
-### Bar chart
+<!--[Source]()-->
```

Comment:
@matthewstern is there a CMAP webpage or PDF that contains this graphic that we can link as the source?

Reply:
I've found several variations of the chart on the website, but not the exact one that you've saved on Dropbox.

Reply:
I'm working on this, I got it directly from Austen so dropbox was my cheap over-the-weekend out. I don't think it's published directly on the website anywhere but I'm hoping we can at least link to a published committee presentation.

Reply:
Just pushed an imperfect fix for this. That link has the most recent CMAP website version at the bottom.